store/mongo/datastore_mongo.go

```diff
@@ -810,50 +758,60 @@ func (db *DataStoreMongo) GetDevCountByStatus(ctx context.Context, status string
 		"$count": "count",
 	}
 
-	var resp bson.M
-
-	pipe := c.Pipe([]bson.M{grp, proj, sum, filt, cnt})
-	err := pipe.One(&resp)
+	var resp []bson.M
 
-	switch err {
-	case nil:
-		break
-	case mgo.ErrNotFound:
-		return 0, nil
-	default:
+	cursor, err := c.Aggregate(ctx, []bson.M{grp, proj, sum, filt, cnt})
```

Comment:
Didn't know you were allowed to use `M` type documents for pipelines (considering the Pipeline typedef in the docs).

Reply:
that was inspired by inventory service: https://github.com/mendersoftware/inventory/blob/a1bca36b3cff2262d5bc9347a9af06b90ae55de5/store/mongo/datastore_mongo.go#L229
https://github.com/mendersoftware/inventory/blob/a1bca36b3cff2262d5bc9347a9af06b90ae55de5/store/mongo/datastore_mongo.go#L508
I think we should use bson.D instead of bson.M in aggregation
I'll change this here and we should probably change it in the inventory service
what do you think @merlin-northern ?

Reply:
There are no inherent requirements for the individual pipeline stages to be ordered in this case so it should be fine. On the other hand, you never know if they suddenly change the argument type for `Aggregate` to `Pipeline`.

Reply:
I think that it should work, to the best of my knowledge, with both D or M, I mean it will return the same set of results. I will perform more tests to day with the standalone code I used for testing/benchmarking with copy of a real-life db, and let you know. if you have some proof that we should use D, then let have it. I only have tests results, manual tests, and what is in the docs. Later today I will perform the actual tests with the pipelines from you in here and from inventory.